File: examples/auth/jwt-persistence.swift
import JsBaoClient

// Opt in to JWT persistence so a relaunch can reuse the short-lived token while
// it is still valid, instead of forcing a fresh sign-in. The token is stored in
// the Keychain under `storageKeyPrefix`. `waitForAuthBootstrap()` restores any
// persisted session before the client is used.
func setupPersistedClient() async throws -> JsBaoClient {
  // #region example
  let client = JsBaoClient(options: JsBaoClientOptions(
    apiUrl: "https://primitiveapi.com",
    wsUrl: "wss://primitiveapi.com",
    appId: "YOUR_APP_ID",
    auth: AuthConfig(
      persistJwtInStorage: true,
      storageKeyPrefix: "my-app"
    )
  ))

  try await client.waitForAuthBootstrap()
  // ["mode": "persisted" | "memory", "prefix": <storageKeyPrefix>]
  let info = client.getAuthPersistenceInfo()
  // #endregion example
  _ = info
  return client
}

File: examples/auth/jwt-persistence.ts
import { initializeClient } from "js-bao-wss-client";

// Opt in to JWT persistence so a reload can reuse the short-lived token while
// it is still valid, instead of forcing a fresh sign-in. The token is cached
// in browser storage under `storageKeyPrefix` (namespace it when several
// clients share an origin).
export async function setupPersistedClient() {
  // #region example
  const client = await initializeClient({
    apiUrl: "https://primitiveapi.com",
    wsUrl: "wss://primitiveapi.com",
    appId: "YOUR_APP_ID",
    auth: {
      persistJwtInStorage: true,
      storageKeyPrefix: "my-app",
    },
  });

  // mode: "memory" | "persisted"; hydrated: was a cached token restored?
  const info = client.getAuthPersistenceInfo();
  // #endregion example
  return { client, info };
}
